R/telemetry.R

```diff
@@ -490,6 +472,165 @@ Telemetry <- R6::R6Class( # nolint object_name_linter
       )
     },
 
+    .log_all_inputs = function(
+      track_values,
+      excluded_inputs,
+      navigation_inputs,
+      session
+    ) {
+      checkmate::assert_r6(session, "ShinySession")
+      checkmate::assert_flag(track_values)
+      checkmate::assert_character(excluded_inputs, null.ok = TRUE)
+      checkmate::assert_character(navigation_inputs, null.ok = TRUE)
+
+      if (is.null(navigation_inputs)) {
+        navigation_inputs <- c()
+      }
+
+      input_values <- shiny::isolate(
+        shiny::reactiveValuesToList(session$input)
+      )
+      session$userData$shiny_input_values <- input_values
+
+      logger::log_debug(logger::skip_formatter(
+        paste(
+          "shiny inputs initialized:",
+          paste(names(input_values), collapse = ", ")
+        )),
+        namespace = "shiny.telemetry"
+      )
+
+      # Log initial value for navigation
+      input_values[names(input_values) %in% navigation_inputs] %>%
+        names() %>%
+        purrr::walk(function(.x) {
+
+          self$log_navigation_manual(
+            navigation_id = .x,
+            value = input_values[[.x]],
+            session = session
+          )
+        })
```

Comment:
Since we are using `purrr`, I think we can make this less verbose by using `~`. 

```
input_values[names(input_values) %in% navigation_inputs] %>%
        names() %>%
        purrr::walk(
          ~ self$log_navigation_manual(
            navigation_id = .x,
            value = input_values[[.x]],
            session = session
          )
        )
```